File: backend/app/schemas/doctor.py
from pydantic import BaseModel
from typing import List, Optional
from uuid import UUID
from datetime import datetime
from .enums import DoctorFeedbackRating, FeedbackCategory


class PatientSummary(BaseModel):
    id: UUID
    user_id: UUID
    full_name: str
    last_triage: Optional[str] = None


class ReportSummary(BaseModel):
    id: UUID
    session_id: UUID
    patient_id: UUID
    patient_name: str
    generated_at: datetime
    urgency_level: str


class FeedbackCreate(BaseModel):
    rating: DoctorFeedbackRating
    correction_text: Optional[str] = None
    feedback_category: Optional[FeedbackCategory] = None

File: frontend/src/types/doctor.ts
import { DoctorFeedbackRating, FeedbackCategory } from './enums';

export interface DoctorDashboardData {
  total_patients: number;
  pending_reports: number;
}

export interface PatientSummary {
  id: string;
  user_id: string;
  full_name: string;
  last_triage: string | null;
}

export interface PatientDetail {
  id: string;
  user_id: string;
  full_name: string;
  date_of_birth: string | null;
  sex: string | null;
  height_cm: number | null;
  weight_kg: number | null;
}

export interface ReportSummary {
  id: string;
  session_id: string;
  patient_id: string;
  patient_name: string;
  generated_at: string;
  urgency_level: string;
}

export interface FeedbackCreate {
  rating: DoctorFeedbackRating;
  correction_text?: string | null;
  feedback_category?: FeedbackCategory | null;
}
